api/services/booking.ts

```diff
@@ -0,0 +1,48 @@
+import { generateBaseQueryKeyFromRoute, routes } from "@/api/routes";
+import { useAppMutation, useAppQuery} from "@/api/client-constructor";
+import { BookingInformationResponse, BookingVerifyPaymentResponse, CustomApiResponse, InitiateBookingPaymentRequest, InitiateBookingRequest, SimpleBookingResponse } from "@/api/types";
+
+
+
+export const getUserBookings = () => {
+    const baseQueryKey = generateBaseQueryKeyFromRoute(routes.bookings.viewUserBookings)
+    return useAppQuery<SimpleBookingResponse>({
+        apiRoute: routes.bookings.viewUserBookings,
+        queryKey: [baseQueryKey]
+    })
+}
+
+export const useGetBookingInformation = (id: string) => {
+    const baseQueryKey = generateBaseQueryKeyFromRoute(routes.bookings.viewAndEditBooking(id))
+    return useAppQuery<BookingInformationResponse>({
+        apiRoute: routes.bookings.viewAndEditBooking(id),
+        queryKey: [baseQueryKey]
+    })
+}
+
+
+
+export const useInitiateBooking = (packageId: string, body?: InitiateBookingRequest) => {
+    return useAppMutation<CustomApiResponse>({
+        apiRoute: routes.bookings.initiateBooking(packageId),
+        method: 'POST',
+        body: JSON.stringify(body)
+    })
+}
+
+
+export const useInitiateBookingPayment = (bookingId: string, body?: InitiateBookingPaymentRequest) => {
+    return useAppMutation<any>({
```

Comment:
Why passing any here, the type here should mimic shape of request type

Reply:
Thanks I'll address this. Using "any" type here is not good practice